exercises/simple-linked-list/src/example/java/SimpleLinkedList.java

```diff
@@ -0,0 +1,79 @@
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+/**
+ *
+ * @author matt
+ */
+public class SimpleLinkedList {
+
+    class Element {
+
+        int value;
+        Element next;
+
+        Element(int value) {
+            this.value = value;
+        }
+    }
+
+    private Element head;
+    private int size = 0;
+
+    public SimpleLinkedList() {
+    }
+
+    public SimpleLinkedList(int[] values) {
+        for (int ii = values.length - 1; ii >= 0; ii--) {
+            push(values[ii]);
+        }
+    }
+
+    final public void push(int value) {
```

Comment:
The warning about calling non-final methods in constructors is conservative.  It's an attempt to enforce the [Open-Closed Principle](https://en.wikipedia.org/wiki/Open/closed_principle).

Specifically, if a method that is being invoked in a constructor is non-final, it technically can be overridden in a subclass.  If someone created a subclass and overrode that method, then the base class' constructor would invoke that overriding implementation (example at: https://gist.github.com/jtigger/61ce1c48543a78846141).